src/edu/ucsb/nceas/metacat/MetacatHandler.java

```diff
@@ -345,448 +320,463 @@ public static InputStream read(String docid, String dataType) throws ParseLSIDEx
-	        docid = DocumentUtil.appendRev(docid);
-	        DocumentImpl doc = new DocumentImpl(docid, false);
-		    // deal with data or metadata cases
-	        if (doc.getRootNodeID() == 0) {
-	            // this is a data file
-	            // get the path to the file to read
-	            try {
-	                String filepath = PropertyService.getProperty("application.datafilepath");
-	                // ensure it is a directory path
-	                if (!(filepath.endsWith("/"))) {
-	                    filepath += "/";
-	                }
-	                String filename = filepath + docid;
-	                inputStream = readFromFilesystem(filename);
-	            } catch (PropertyNotFoundException pnf) {
-	                logMetacat.debug("There was a problem finding the "
-	                        + "application.datafilepath property. The error "
-	                        + "message was: " + pnf.getMessage());
-	                throw pnf;
-	            } // end try()
-	        } else {
-	            // this is an metadata document
-	            // Get the xml (will try disk then DB)
-	            try {
-	                // force the InputStream to be returned
-	                OutputStream nout = null;
-	                inputStream = doc.toXml(nout, null, null, true);
-	            } catch (McdbException e) {
-	                // report the error
-	                logMetacat.error(
-	                        "MetacatHandler.readFromMetacat() "
-	                                + "- could not read document " + docid + ": "
-	                                + e.getMessage(), e);
-	            }
-	        }
-		}
-		return inputStream;
-	}
-    
+        } else {
+            // accommodate old clients that send docids without revision numbers
+            docid = DocumentUtil.appendRev(docid);
+            DocumentImpl doc = new DocumentImpl(docid, false);
+            // deal with data or metadata cases
+            if (doc.getRootNodeID() == 0) {
+                // this is a data file
+                // get the path to the file to read
+                try {
+                    String filepath = PropertyService.getProperty("application.datafilepath");
+                    // ensure it is a directory path
+                    if (!(filepath.endsWith("/"))) {
+                        filepath += "/";
+                    }
+                    String filename = filepath + docid;
+                    inputStream = readFromFilesystem(filename);
+                } catch (PropertyNotFoundException pnf) {
+                    logMetacat.debug("There was a problem finding the "
+                                         + "application.datafilepath property. The error "
+                                         + "message was: " + pnf.getMessage());
+                    throw pnf;
+                } // end try()
+            } else {
+                // this is n metadata document
```

Comment:
Should it be an or n?

Reply:
It should be "a metadata document". The difference between `a` and `an` is about the pronunciation of the following word -- typically use `a` unless the following word starts with a vowel sound, in which case use `an` (e.g., "an action", "an egg", "an hour",  but "a house"). See: https://english.stackexchange.com/a/155